gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/MFAChallengeStep.java

```diff
@@ -42,13 +52,15 @@ public void execute(RoutingContext routingContext, AuthenticationFlowChain flow)
         final Session session = routingContext.session();
         var context = new MfaFilterContext(client, session);
 
+        final Map<String, Object> data = getEvaluableAttributes(routingContext);
+
         // Rules that makes you skip MFA challenge
         var mfaFilterChain = new MfaFilterChain(
                 new ClientNullFilter(client),
                 new NoFactorFilter(client.getFactors()),
                 new MfaChallengeCompleteFilter(session),
-                new AdaptiveMfaFilter(context, ruleEngine, routingContext.request(), routingContext.data()),
-                new StepUpAuthenticationFilter(context, ruleEngine, routingContext.request(), routingContext.data()),
+                new AdaptiveMfaFilter(context, ruleEngine, routingContext.request(), data),
```

Comment:
I am wondering if we should put the routingContext in the MFAFilterContext, and have a `getEvaluableAttributes ` inside ?

🤔